**1. Trabajo Base De Datos**


In [5]:

if (!require(wooldridge)) install.packages("wooldridge")
library(wooldridge)

data("ceosal1")
ceosal1 <- na.omit(ceosal1)

**1.1. (10 puntos) Examine la base y elabore una breve descripción de sus variables, distinguiendo
entre dependientes, explicativas, continuas y dummies. ¿Para qué tipo de estudio económico
se emplea habitualmente esta base de datos?**



In [4]:
set.seed(1)
data(ceosal1)
ceosal1 <- na.omit(ceosal1)
?ceosal1

In [7]:
print(names(ceosal1))
print(str(ceosal1))

 [1] "salary"   "pcsalary" "sales"    "roe"      "pcroe"    "ros"     
 [7] "indus"    "finance"  "consprod" "utility"  "lsalary"  "lsales"  
'data.frame':	209 obs. of  12 variables:
 $ salary  : int  1095 1001 1122 578 1368 1145 1078 1094 1237 833 ...
 $ pcsalary: int  20 32 9 -9 7 5 10 7 16 5 ...
 $ sales   : num  27595 9958 6126 16246 21783 ...
 $ roe     : num  14.1 10.9 23.5 5.9 13.8 ...
 $ pcroe   : num  106.4 -30.6 -16.3 -25.7 -3 ...
 $ ros     : int  191 13 14 -21 56 55 62 44 37 37 ...
 $ indus   : int  1 1 1 1 1 1 1 1 1 1 ...
 $ finance : int  0 0 0 0 0 0 0 0 0 0 ...
 $ consprod: int  0 0 0 0 0 0 0 0 0 0 ...
 $ utility : int  0 0 0 0 0 0 0 0 0 0 ...
 $ lsalary : num  7 6.91 7.02 6.36 7.22 ...
 $ lsales  : num  10.23 9.21 8.72 9.7 9.99 ...
 - attr(*, "time.stamp")= chr "25 Jun 2011 23:03"
NULL


La base de datos ceosal1 viene del libro wooldridge, el cual se tomo una muestra aleatoria que contiene 209 observaciones y 12 variables relacionadas con el salario y...




### 1.1
La base de datos ceosal1, proveniente del libro de Jeffrey Wooldridge, corresponde a una muestra aleatoria de 209 observaciones y 12 variables, recopiladas a partir de información publicada en BusinessWeek en 1991.

Esta base es ampliamente utilizada en estudios de economía laboral y gobierno corporativo, ya que permite analizar los factores que influyen en la determinación de los salarios de altos ejecutivos

| Variable   | Tipo      | Rol                     | Descripción                                 |
|------------|-----------|-------------------------|---------------------------------------------|
| `salary`   | Continua  | **Dependiente**         | Salario del CEO en 1990 (miles de USD)      |
| `lsalary`  | Continua  | Dependiente alternativa | log(salary)                                 |
| `pcsalary` | Continua  | Dependiente alternativa | % cambio en salario 1989–1990               |
| `sales`    | Continua  | **Explicativa**         | Ventas empresa 1990 (millones USD)          |
| `roe`      | Continua  | **Explicativa**         | Retorno sobre patrimonio promedio 1988–1990 |
| `ros`      | Continua  | **Explicativa**         | Retorno sobre acciones 1988–1990            |
| `lsales`   | Continua  | Explicativa             | log(sales)                                  |
| `pcroe`    | Continua  | Explicativa             | % cambio en ROE 1988–1990                   |
| `indus`    | **Dummy** | Control sectorial       | =1 si empresa industrial                    |
| `finance`  | **Dummy** | Control sectorial       | =1 si empresa financiera                    |
| `consprod` | **Dummy** | Control sectorial       | =1 si bienes de consumo                     |
| `utility`  | **Dummy** | Control sectorial       | =1 si transporte/servicios públicos         |



**1.2. (5 puntos) Genere una variable indicadora (dummy) que tome el valor 1 cuando la variable
sales supere su promedio y 0 en caso contrario. Asigne a esta nueva variable un nombre que
sea fácilmente identificable**


In [12]:
ceosal1$sales_mayor_promedio <- ifelse(ceosal1$sales > mean(ceosal1$sales), 1, 0)
cat("Promedio de sales:", round(mean(ceosal1$sales), 2), "millones USD\n")
table(ceosal1$sales_mayor_promedio)

Promedio de sales: 6923.79 millones USD



  0   1 
155  54 

##1.2
Se crea sales_mayor_promedio, que identifica con el numero 1 empresas con ventas sobre el promedio y con 0 las que no

**1.3. (5 puntos) Cree una variable de la multuplicación entre la variable dummy antes creada y la
variable indus. ¿Qué indica esta interacción?**

In [13]:
ceosal1$sales_mayor_promedio_indus <- ceosal1$sales_mayor_promedio * ceosal1$indus
table(ceosal1$sales_mayor_promedio_indus)


  0   1 
188  21 

##1.3
Es 1 cuando la empresa esta dentro del sector industrial y tiene ventas mayores que el promedio.


**1.4. (5 puntos) Cree una variable de la multiplicación entre la variable continua roe y la variable
dummy antes creada. ¿Qué indica esta interacción?**


In [16]:
ceosal1$sales_mayor_promedio_roe <- ceosal1$sales_mayor_promedio * ceosal1$roe
summary(ceosal1$sales_mayor_promedio_roe)

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  0.000   0.000   0.000   4.251   3.500  43.700 

##1.4
sales_mayor_promedio_roe: Representa el efecto del retorno sobre e roe específicamente para aquellas empresas con ventas superiores al promedio. Permite ver si el mercado premia el desempeño de forma distinta en empresas grandes.

**1.5. (5 puntos) Cree una variable que indique si el compañia corresponde a industria (indus) ó si
finance. Asigne a esta nueva variable un nombre que sea fácilmente identificable**

In [23]:
ceosal1$indus_o_finance <- ceosal1$indus + ceosal1$finance
table(ceosal1$indus_o_finance)


  0   1 
 96 113 

##1.5
Dado que indus y finance no pueden tomar valor 1 al mismo tiempo, al sumarlas se obtiene una dummy que vale 1 si la empresa pertenece a alguno de esos sectores y 0 en caso contrario, sin posibilidad de duplicidad

**2. Modelos de Regresión Lineal**

**2.1. (15 puntos) Genere un modelo de regresión lineal, sobre alguna de las variables dependientes,
con respecto a las variables sales, roe y ros. Interprete sus resultados y la significancia de los
regresores. ¿El modelo tiene significancia global al 5%?**



In [18]:
modelo <- lm(salary ~ sales + roe + ros , data = ceosal1)
summary (modelo)



Call:
lm(formula = salary ~ sales + roe + ros, data = ceosal1)

Residuals:
    Min      1Q  Median      3Q     Max 
-1495.0  -486.5  -242.8    98.9 13516.4 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept) 864.117516 228.399737   3.783 0.000203 ***
sales         0.015482   0.008954   1.729 0.085291 .  
roe          22.003312  11.516554   1.911 0.057454 .  
ros          -1.105191   1.450240  -0.762 0.446892    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1360 on 205 degrees of freedom
Multiple R-squared:  0.03191,	Adjusted R-squared:  0.01775 
F-statistic: 2.253 on 3 and 205 DF,  p-value: 0.08337


##2.1
Intercepto: Si las ventas, el ROE y el ROS fueran cero, el salario base estimado sería de $864,118 aproximadamente.  

Sales: Por cada millón de dólares adicionales en ventas, el salario del CEO aumenta en $15.48. Es un efecto positivo, pero pequeño.  

RO: Por cada punto porcentual de aumento en el retorno sobre el capital, el salario aumenta en $22,003.  

ROS: Sugiere que un mayor retorno sobre las acciones está asociado a una leve baja en el salario, aunque veremos que esto no es estadísticamente relevante

**2.2. (10 puntos) Al modelo de regresión lineal anterior, agregue las variables creadas en el inciso 1.2
y 1.4, interprete sus resultados. ¿Cómo afectan los regresores de estas variables en el salario?**


In [19]:
modelo_22 <- lm(salary ~ sales + roe + ros + sales_mayor_promedio +
                sales_mayor_promedio_roe, data = ceosal1)
summary(modelo_22)


Call:
lm(formula = salary ~ sales + roe + ros + sales_mayor_promedio + 
    sales_mayor_promedio_roe, data = ceosal1)

Residuals:
    Min      1Q  Median      3Q     Max 
 -976.0  -469.1  -263.4    97.2 13566.3 

Coefficients:
                           Estimate Std. Error t value Pr(>|t|)   
(Intercept)              783.298711 258.064395   3.035  0.00272 **
sales                      0.006657   0.011197   0.594  0.55285   
roe                       24.472338  13.088222   1.870  0.06295 . 
ros                       -0.986904   1.457818  -0.677  0.49919   
sales_mayor_promedio     513.083213 510.134319   1.006  0.31572   
sales_mayor_promedio_roe  -9.498301  25.587585  -0.371  0.71087   
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1361 on 203 degrees of freedom
Multiple R-squared:  0.04059,	Adjusted R-squared:  0.01696 
F-statistic: 1.718 on 5 and 203 DF,  p-value: 0.132


##2.2
Al incluir la variable dummy y su interacción con el ROE, el poder explicativo del modelo ($R^2$) mejora levemente, pero ninguna de las nuevas variables es significativa al 5%. Esto sugiere que el hecho de que una empresa tenga ventas sobre el promedio no cambia de manera estadísticamente significativa la estructura de compensación del CEO en esta muestra, y el modelo en su conjunto sigue sin ser capaz de explicar el salario de forma robusta ($p=0.132$)

**2.3. (15 puntos) Realiza un test de hipotesis para evaluar si las dos variables que se agregaron son
significativas para el modelo simultaneamente. Use significancia del 5%**


In [20]:

R2_r  <- summary(modelo_21)$r.squared
R2_ur <- summary(modelo_22)$r.squared
n     <- nrow(ceosal1)
q     <- 2
k_ur  <- 5

F_critico <- qf(0.95, q, n - k_ur - 1)
F_est     <- (R2_ur - R2_r) / q / ((1 - R2_ur) / (n - k_ur - 1))

cat("R² restringido:     ", round(R2_r, 5), "\n")
cat("R² no restringido:  ", round(R2_ur, 5), "\n")
cat("F estimado:         ", round(F_est, 4), "\n")
cat("F crítico (5%):     ", round(F_critico, 4), "\n")
if (F_est > F_critico) {"Se rechaza H0"} else {"No se rechaza H0"}

R² restringido:      0.03191 
R² no restringido:   0.04059 
F estimado:          0.9178 
F crítico (5%):      3.0404 


[1] "No se rechaza H0"

##2.3
Para evaluar si las dos variables añadidas (sales_mayor_promedio y sales_mayor_promedio_roe) son significativas simultáneamente al modelo, planteamos las siguientes hipótesis:

$H_0: \beta_{dummy} = 0 \text{ y } \beta_{interacción} = 0$ ).  
$H_1: \text{Al menos uno de los coeficientes es distinto de cero}$.  

F estimado: 0.9178  F crítico (5%): 3.0404

Decision
Si $F_{estimado} > F_{crítico}$, se rechaza $H_0$.  En este caso, $0.9178 < 3.0404$, por lo tanto, No se rechaza la hipótesis nula ($H_0$).

**2.4.  (15 puntos) Volviendo a la regresión del salario en función de sales, roe y ros. Considere
dos grupos, Grupo 1: Aquellos clientes que sales es mayor al promedio, y Grupo 2: Aquellos
clientes que sales es menor al promedio. Realice un test de hipotesis para grupos al 5% de
significancia. Use el metodo que más le acomode**

In [21]:
ceosal1$sales_D <- ceosal1$sales * ceosal1$sales_mayor_promedio
ceosal1$roe_D   <- ceosal1$roe   * ceosal1$sales_mayor_promedio
ceosal1$ros_D   <- ceosal1$ros   * ceosal1$sales_mayor_promedio

modelo_24 <- lm(salary ~ sales + roe + ros + sales_mayor_promedio +
                sales_D + roe_D + ros_D, data = ceosal1)
summary(modelo_24)


R2_r  <- summary(modelo_21)$r.squared
R2_ur <- summary(modelo_24)$r.squared
n     <- nrow(ceosal1)
q     <- 4
k_ur  <- 7

F_critico <- qf(0.95, q, n - k_ur - 1)
F_est     <- (R2_ur - R2_r) / q / ((1 - R2_ur) / (n - k_ur - 1))

cat("R² restringido:     ", round(R2_r, 5), "\n")
cat("R² no restringido:  ", round(R2_ur, 5), "\n")
cat("F estimado:         ", round(F_est, 4), "\n")
cat("F crítico (5%):     ", round(F_critico, 4), "\n")
if (F_est > F_critico) {"Se rechaza H0"} else {"No se rechaza H0"}


Call:
lm(formula = salary ~ sales + roe + ros + sales_mayor_promedio + 
    sales_D + roe_D + ros_D, data = ceosal1)

Residuals:
    Min      1Q  Median      3Q     Max 
-1024.7  -456.7  -210.2    83.2 13712.4 

Coefficients:
                       Estimate Std. Error t value Pr(>|t|)  
(Intercept)           331.86179  352.76387   0.941   0.3480  
sales                   0.13362    0.06635   2.014   0.0454 *
roe                    25.63426   13.07068   1.961   0.0512 .
ros                    -0.47570    1.66943  -0.285   0.7760  
sales_mayor_promedio 1043.02106  578.05672   1.804   0.0727 .
sales_D                -0.13033    0.06731  -1.936   0.0542 .
roe_D                 -17.86857   28.02885  -0.638   0.5245  
ros_D                   1.55160    3.84565   0.403   0.6870  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1353 on 201 degrees of freedom
Multiple R-squared:  0.06028,	Adjusted R-squared:  0.02756 
F-statistic: 1.842 on 7 and 201

R² restringido:      0.03191 
R² no restringido:   0.06028 
F estimado:          1.517 
F crítico (5%):      2.4166 


[1] "No se rechaza H0"

##2.4
Como el F estimado (1.517) es menor que el F crítico (2.4166), el resultado es "No se rechaza H0".  

Al nivel de significancia del 5%, no hay evidencia de que existan diferencias estructurales en la compensación de los CEOs entre empresas con ventas sobre el promedio y aquellas bajo el promedio. Es decir, a pesar de que el $R^2$ subió al 6%, la mejora no es estadísticamente lo suficientemente fuerte como para decir que los dos grupos deben modelarse de forma separada. La forma en que las ventas, el ROE y el ROS afectan al salario es, en promedio, igual para ambos tipos de empresas

**2.5. (15 puntos) Usando la regresión del salario en función de sales, roe y ros, se distinguen los
grupos mediante las dummies; indus, finance, consprod y utility, que corresponden al tipo de
empresa. Realice mediante test de Chow, con 5% de significancia, si los modelos para cada
grupo son iguales**


In [22]:
ceosal1_indus    <- subset(ceosal1, indus==1)
ceosal1_finance  <- subset(ceosal1, finance==1)
ceosal1_consprod <- subset(ceosal1, consprod==1)
ceosal1_utility  <- subset(ceosal1, utility==1)

modelo_pool     <- lm(salary ~ sales + roe + ros, data = ceosal1)
modelo_indus    <- lm(salary ~ sales + roe + ros, data = ceosal1_indus)
modelo_finance  <- lm(salary ~ sales + roe + ros, data = ceosal1_finance)
modelo_consprod <- lm(salary ~ sales + roe + ros, data = ceosal1_consprod)
modelo_utility  <- lm(salary ~ sales + roe + ros, data = ceosal1_utility)

SSR_r  <- sum(residuals(modelo_pool)^2)
SSR_ur <- sum(residuals(modelo_indus)^2)   +
          sum(residuals(modelo_finance)^2)  +
          sum(residuals(modelo_consprod)^2) +
          sum(residuals(modelo_utility)^2)

G   <- 4
k   <- 3
n   <- nrow(ceosal1)
gl1 <- (G - 1) * (k + 1)
gl2 <- n - G * (k + 1)

F_critico <- qf(0.95, gl1, gl2)
F_est     <- (SSR_r - SSR_ur) / SSR_ur * gl2 / gl1

cat("n:", n, "| gl1:", gl1, "| gl2:", gl2, "\n")
cat("SSR restringido:    ", round(SSR_r, 2), "\n")
cat("SSR no restringido: ", round(SSR_ur, 2), "\n")
cat("F estimado:         ", round(F_est, 4), "\n")
cat("F crítico (5%):     ", round(F_critico, 4), "\n")
if (F_est > F_critico) {"Se rechaza H0"} else {"No se rechaza H0"}

n: 209 | gl1: 12 | gl2: 193 
SSR restringido:     379231123 
SSR no restringido:  348896652 
F estimado:          1.3983 
F crítico (5%):      1.8026 


[1] "No se rechaza H0"

##2.5
Planteamiento de hipotesis

$H_0$: Los coeficientes son iguales para todos los sectores (el modelo es estable).  $H_1$: Al menos uno de los sectores tiene una estructura salarial distinta

Como el F estimado (1.3983) es menor que el F crítico (1.8026), el resultado es "No se rechaza H0".  

Al 5% de significancia, no existe evidencia estadística para afirmar que el sector de la empresa afecte la estructura salarial de los CEOs por lo que os modelos para cada grupo son estadísticamente iguales.  

A pesar de las diferencias obvias entre el sector financiero y el industrial, en esta base de datos (ceosall), esas diferencias no se traducen en una forma distinta de calcular la compensación de sus directivos